From 42d9186f9ef41d6b50458db13ca34d01595e1ecd Mon Sep 17 00:00:00 2001
From: Mike Pagano <mpagano@gentoo.org>
Date: Wed, 20 Jun 2018 12:31:18 -0400
Subject: [PATCH] kvmclock: Define pvclock_pvti_cpu0_va setter for X86_32
Cc: mpagano@gentoo.org

setup_vsyscall_timeinfo() is only defined for x86_64, thus
vclock_set_pvti_cpu0_va() does not get called resulting in
the failure of ptp_kvm initialization for Linux X86_32 guests.
The result of this being that the 32 bit guest userspace has
no /dev/ptp0 device.

See Gentoo bug 658544 located at the following link:
https://bugs.gentoo.org/658544

Signed-off-by: Mike Pagano <mpagano@gentoo.org>
Signed-off-by: Andreas Steinmetz <ast@domdv.de>
---
 arch/x86/kernel/kvmclock.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/arch/x86/kernel/kvmclock.c b/arch/x86/kernel/kvmclock.c
index bf8d1eb7fca3..6aee5c6265b3 100644
--- a/arch/x86/kernel/kvmclock.c
+++ b/arch/x86/kernel/kvmclock.c
@@ -350,7 +350,7 @@ void __init kvmclock_init(void)
 
 int __init kvm_setup_vsyscall_timeinfo(void)
 {
-#ifdef CONFIG_X86_64
+#ifdef CONFIG_X86_64 || defined(CONFIG_X86_32)
 	int cpu;
 	u8 flags;
 	struct pvclock_vcpu_time_info *vcpu_time;
-- 
2.16.4

